### DISCUSSION TITLE
Is there a way to get insights into a Seq application?

### DISCUSSION THREAD

**Discussion by per-samuelsson** [datalust/seq-tickets#1994] [Help] [answered]
Hi,

I've been evaluating [Seq.Apps.CertifcateCheck](https://github.com/janpieterz/seq-input-certificatecheck) for some time now, and suddenly, it has just stopped reporting about any expiring certificate.

I tested the configuration by copying it (via JSON) to a local Seq-instance, and it immediately did what it should.

What I have tried looking for (through logs on the server, under `Settings | Diagnostics`, etc) is a way to get some insight into this application. Is it even called by Seq? How often? Anything that could help me get a better understanding of what might be broken.

1. Is there some way to do the above?
2. Is there some way to restart an application? I haven't seen any documentation on application lifetime management as I recall. I guess changing the configuration would do the trick, but seem hacky.

Edit: Oh, and it doesn't report any diagnostic events about errors either.

**Comment by nblumhardt** [comment, marked as answer]
Hi @per-samuelsson,

I just had a quick scan through the code and spotted one possible issue - exceptions raised during certificate checking won't release the lock protecting the shared HTTP client:

https://github.com/janpieterz/seq-input-certificatecheck/blob/main/Seq.Input.CertificateCheck/HttpClientWrapper.cs#L61C34-L61C34

I'll raise an issue over there, the code change is reasonably recent, I think.

The other option for debugging are:

- use `seqcli app host` to load up the assemblies from a downloaded copy of the NuGet package, or
- create an instance of the app in copy of Seq running on your developer machine, and attach a debugger to the `seqcli` child process to see what's going on in the app.

Hope this helps,
Nick

**Reply by per-samuelsson** [reply]
Wow @nblumhardt , that's some great support right there for sure. 🥇 Less than 10 minutes after I raised the question.

Thanks a lot, I'll follow that issue then, and will keep those tips about debugging for later.

BTW, on a similar topic for others that possibly come here and would like to debug their own applications interactively, I'll take the oppurtinity to mention the great blog post that I had lots of use for during development of an app of my own. It worked like a charm and was very valuable.

https://blog.datalust.co/debugging-seq-apps

**Reply by nblumhardt** [reply]
Thanks @per-samuelsson! 👍
